book/src/linking_native_functions.md

```diff
@@ -0,0 +1,96 @@
+# Linking native functions into the JVM
+
+Using the [`#[java_function]`](./java_function.md) decorator you can write Rust implementations for Java native functions. To get the JVM to invoke these methods, it has to know how to find them. The way you do this depends on whether you have the "top-level" program is written in Rust or in Java.
+
+## Rust program that creates a JVM
+
+If your Rust program is launching the JVM, then you can configure that JVM to link to your native method definitions through methods on the JVM builder.
+
+```rust
+use duchess::prelude::*; // 👈 You'll need this.
+
+#[java_function(...)]
+fn foo(...) { }
+
+fn main() -> duchess::GlobalResult<()> {
+    Jvm::builder()
+        .link(foo::java_fn()) // 👈 Note the `::java_fn()`.
+        .try_launch()?;
+}
+```
+
+**How it works.** The call `foo::java_fn()` returns a `duchess::JavaFunction` struct. The `java_fn` method is defined in the duchess `JavaFn` trait; that trait is implemented on a struct type `foo` that is created by the `#[java_function]` decorator. This trait is in the duchess prelude, which is why you need to `use duchess::prelude::*`.
+
```

Comment:
Linking the same function multiple times seems to work fine. If it's intended, should this be mentioned in the documentation?